src/app/collections/Collections.tsx

```diff
@@ -48,25 +52,29 @@ function mapStateToProps() {
   return (state: RootState): StoreProps => ({
     buckets: bucketsSelector(state),
     defs: state.manifest.d2Manifest,
+    stores: storesSelector(state),
     ownedItemHashes: ownedItemHashesSelector(state),
     profileResponse: profileResponseSelector(state),
   });
 }
 
-const refreshStores = () =>
-  refresh$.subscribe(() => {
-    D2StoresService.reloadStores();
-  });
+function getStoresService(account: DestinyAccount) {
+  return account.destinyVersion === 1 ? D1StoresService : D2StoresService;
+}
 
 /**
  * The collections screen that shows items you can get back from the vault, like emblems and exotics.
  */
-function Collections({ account, buckets, ownedItemHashes, defs, profileResponse }: Props) {
+function Collections({ account, stores, buckets, ownedItemHashes, defs, profileResponse }: Props) {
   useEffect(() => {
-    D2StoresService.getStoresStream(account);
-  }, [account]);
+    if (!stores.length) {
+      getStoresService(account).getStoresStream(account);
+    }
+  });
```

Comment:
This needs a dependencies array (I'm surprised eslint let you get away without it) - it should depend on `[stores.length, account]`.

Reply:
ah - hmm, I'm seeing something similar in the organize, should I update it there as well?

https://github.com/DestinyItemManager/DIM/blob/d35cda5aa7886d07083b74fe5cb79e5f48931aa4/src/app/organizer/Organizer.tsx#L50-L54

Inventory works similar-ish. Can chat offline on how to wrap these maybe in destiny.tsx to not duplicate this logic across each sub-page
https://github.com/DestinyItemManager/DIM/blob/f67ed465fd7e7bff57f650949a6e4fc7d0a1672a/src/app/inventory/Inventory.tsx#L48-L64

Reply:
iirc it'll yell at you for incomplete arrays but assumes a nonexistent one is intentional?
weirdly i think it doesn't allow an empty one.

Reply:
Ah - the inventory one is correct. Let's just lift that into a "useLoadStores" hook.

Reply:
I created the hook and am using it in a component that gets used in Destiny.tsx. I'm fine w/ just using the hook in each place we need it, but then we'd need to have each view implement the loading view as it does currently instead of now just the Destiny.tsx file handling it. Not sure on the best way to implement this, happy to make any other changes